framework/test_app/runners/k8s/k8s_base_runner.py

```diff
@@ -53,15 +58,17 @@ class RunHistory:
     time_stopped: _datetime
 
 
+# TODO(sergiitk): Wow this class is dirty. Tracking managed object state mixed
+#   with k8s API wrappers, yaml parsing, mako templating and other weird stuff.
+#   Public methods are impossible to find among all the _private().
+#   This monster must be broken apart, or it'll eat us first.
```

Comment:
👹 This reads like the last log entry found in the captain's journal on a ghost ship.

Reply:
This is pretty much how I felt writing it 😆